cofacts/rumors-db
# 201712 mappings refactor

Related to #7

The new mapping: [201712 version](https://hackmd.io/s/BkxbQ8ZbM)

This PR:
- Enables DB migrations, which can be done using these steps : https://github.com/cofacts/rumors-db/issues/7#issuecomment-352262625
- Adds schema versioning by appending package.json's `version` after each index names, and add alias to the versioned index (see `loadSchema.js`).
- Add a `docker-compose.yml` so that we can use `docker-compose up` to spin up a database to test the schema.

## Steps to verify

1. `docker-compose up` in rumors-db's directory
2. Follow the steps in https://github.com/cofacts/rumors-db/issues/7#issuecomment-352262625 (You may need to contact @mrorz for help to get the current production DB snapshot)
3. Enter kibana on `localhost:62223` and look around the new indexes to see if anything is not expected.

## Changes to take care of

There are lots of changes in this PR. Most of them are utilities executed many times, so that we can assume they are correct. However, the migration scripts, especially `db/migrations/201712-001-reindex.js`, is really prone to have logic errors, and thus requires reviewers' attention.

## TODO (beyond the scope of this PR)
- Remove unused `data/` directory and `scripts/` directory, and their relative paragraphs in README.md
- Make changes in `rumors-deploy`'s docker-compose so that it aligns with this PR's `docker-compose`
